File: problems/67/solution.go
func addBinary(a string, b string) string {
    an, bn, res, carry := len(a) - 1, len(b) - 1, "", false
    
    for an >= 0 || bn >= 0 || carry {
        nch := 0
        
        if carry { nch++ }
        if an >= 0 && a[an] == '1' { nch++ }
        if bn >= 0 && b[bn] == '1' { nch++ }
        
        an--
        bn--
        carry = nch >= 2
        nch %= 2
        
        res = strconv.Itoa(nch) + res
    }
    
    return res
}

File: problems/67/solution.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        overhead = False
        i = len(a)
        j = len(b)
        res = ''
        
        while i > 0 or j > 0:
            i -= 1
            j -= 1
            ach = a[i] if i >= 0 else '0'
            bch = b[j] if j >= 0 else '0'
            ap = None
            
            if ach == '1' and ach == bch:
                ap = '1' if overhead else '0'
                overhead = True
            elif ach == '0' and ach == bch:
                ap = '1' if overhead else '0'
                overhead = False
            else:
                ap = ach if ach == '1' else bch
                
                if overhead:
                    ap = '0'
            
            res = ap + res
        
        if overhead:
            res = '1' + res
        
        return res